### DISCUSSION TITLE
BitConverter and Endian-ness

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#64953] [Q&A] [answered]
Is there a way to specify the endianness of the bytes that you are providing/requesting to/from bitconverter?

I am interfacing with a legacy file format and want my code to work without having to flip glob based on the source systems endianness. It would be really nice if there was an enum I could Pass in to specify the format.

**Comment by stephentoub** [comment, marked as answer]
Have you looked at BinaryPrimitives?

**Reply by TonyValenti** [reply]
@stephentoub - That was exactly what I needed!  Thank you so much!
Seems like BitConverter would be a much more discoverable place for those methods.

**Reply by khellang** [reply]
There's also a proposal for endianness parameters for `BinaryReader`/`BinaryWriter`; https://github.com/dotnet/runtime/issues/26904.
